recipe/activate.sh

```diff
@@ -1,7 +1,30 @@
 #!/bin/bash
 
-if [[ -n "$CXX" ]]; then
-    export NVCC_PREPEND_FLAGS_BACKUP="${NVCC_PREPEND_FLAGS}"
-    export NVCC_PREPEND_FLAGS="${NVCC_PREPEND_FLAGS} -ccbin=${CXX}"
+export CFLAGS_CUDA_NVCC_BACKUP="${CFLAGS}"
+export CPPFLAGS_CUDA_NVCC_BACKUP="${CPPFLAGS}"
+export CXXFLAGS_CUDA_NVCC_BACKUP="${CXXFLAGS}"
+export NVCC_PREPEND_FLAGS_BACKUP="${NVCC_PREPEND_FLAGS}"
```

Comment:
Would suggest only backing these up if they are set and skipping if not (like so)

```suggestion
if [[ ! -z "${CFLAGS+x}" ]]
then
  export CFLAGS_CUDA_NVCC_BACKUP="${CFLAGS}"
fi

if [[ ! -z "${CPPFLAGS+x}" ]]
then
  export CPPFLAGS_CUDA_NVCC_BACKUP="${CPPFLAGS}"
fi

if [[ ! -z "${CXXFLAGS+x}" ]]
then
  export CXXFLAGS_CUDA_NVCC_BACKUP="${CXXFLAGS}"
fi

if [[ ! -z "${NVCC_PREPEND_FLAGS +x}" ]]
then
  export NVCC_PREPEND_FLAGS_BACKUP="${NVCC_PREPEND_FLAGS}"
fi
```